src/app/services/project_service.js

```diff
@@ -280,18 +269,49 @@ angular
             })
         }
 
-        var res = [];
-        _.each(service.project.searchable, function(model) {
-            var matches = fuzzySearchObj(q, model);
-            if (matches.length) {
-                res.push({
-                    model: model,
-                    matches: matches,
-                });
-            }
-        });
-        return res;
+		try {
+			//As search terms become longer, be less tolerant of tiny fuzzy matches
+			var shortestWord = searchTerm.split(' ').sort(function(a, b){ return a.length - b.length})[0]
+			service.fuse.options.minMatchCharLength = Math.max(1, shortestWord.length - 2);
+			
+			const result = service.fuse.search(searchTerm)
+			
+			return _.map(result, function(res) {
+				return {
+					model: res.item,
+					matches: res.matches,
+					overallWeight: res.score * -1 //hack to get sorting right for the moment
+				}
+			});
+		}
+		catch (e) {
+			console.error(`Error searching with fuse: ${e}`);
+		}
+    }
+    
+    function prepareModelsForSearching(models) {
+    	//To tell fuse which fields to search, 
+    	//columns needs to be an array to avoid 
+    	//naming each object individually.
+    	var modelList = _.map(models, function(model) {
+		  // make a copy
+		  var newModel = _.assign({}, model);
+		  newModel.columns = _.values(model.columns);
+		  newModel.searchableName = getModelName(model);
```

Comment:
An interesting side-effect of this approach is that macros from other packages become much surfaced much more often than before (esp on short search terms), e.g. dbt-utils and codegen, because the package name is also passed in as a search option, as opposed to just for disambiguation before being rendered on the page. 

I've done it this way because fuse gives back a range of matching characters for highlighting, and tacking on an extra string afterwards makes that a bit less elegant than I'd like. I think this is somewhere between a net positive and neutral, but might be lying to myself so that I don't have to fix it. Open to alternative views.

In general, I'm a bit bemused at the number of different ways different resources can be named, and that there isn't a canonical field for this. Or is there?

Reply:
haha - ok - this one sounds like a can of worms. dbt does have "unique identifiers" but they are not semantically super useful (eg. they look like `model.fishtown.fct_users`). It would be really good to give every node type a canonical name for search representation, but that's something we'd need to layer in as you've done here AFAIK

Going to come back to this to take a deeper dive soon